server/src/main/java/org/elasticsearch/search/rank/RankBuilder.java

```diff
@@ -68,14 +71,27 @@ public int windowSize() {
     }
 
     /**
-     * Generates a context used to execute required searches on the shard.
+     * Generates a context used to execute required searches during the query phase on the shard.
      */
-    public abstract RankShardContext buildRankShardContext(List<Query> queries, int from);
+    public abstract QueryPhaseRankShardContext buildQueryPhaseShardContext(List<Query> queries, int from);
 
     /**
-     * Generates a context used to perform global ranking on the coordinator.
+     * Generates a context used to be executed on the coordinating node, that would combine all individual shard results.
      */
-    public abstract RankCoordinatorContext buildRankCoordinatorContext(int size, int from);
+    public abstract QueryPhaseRankCoordinatorContext buildQueryPhaseCoordinatorContext(int size, int from);
+
+    /**
+     * Generates a context used to execute the rank feature phase on the shard. This is responsible for retrieving any needed
+     * feature data, and passing them back to the coordinator through the appropriate {@link  RankShardResult}.
+     */
+    public abstract RankFeaturePhaseRankShardContext buildRankFeaturePhaseShardContext();
```

Comment:
These references are a bit out of context now, as the `RankFeature` phase is to be added in a separate PR. 

If you think is best, I can remove these references for now, and add them in the other PR.

Reply:
@pmpailis lets remove these unused pieces of code. No reason to have them IMO as they aren't gonna save much code churn in the later work.